Tests/FuzzilliTests/LifterTest.swift

```diff
@@ -71,13 +71,132 @@ class LifterTests: XCTestCase {
         XCTAssertEqual(prettyCode, expectedPrettyCode)
         XCTAssertEqual(minifiedCode, expectedMinifiedCode)
     }
+
+    func testNestedCodeStrings(){
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+
+        let code = b.codeString() {
+            let v0 = b.loadInt(1337)
+            let v1 = b.loadFloat(13.37)
+            let _ = b.binary(v1, v0, with: .Mul)
+            let _ = b.codeString() {
+                let v0 = b.loadInt(1337)
+                let v1 = b.loadFloat(13.37)
+                let _ = b.binary(v1, v0, with: .Add)
+                let _ = b.codeString() {
+                    let start = b.loadInt(0)
+                    let end = b.loadInt(2)
+                    let step = b.loadInt(1)
+                    b.forLoop(start, .lessThan, end, .Add, step) { _ in
+                        let v1 = b.loadInt(1337)
+                        let _ = b.phi(v1)
+
+                        let _ = b.codeString() {
+                            let v3 = b.loadString("hello world")
+                            let _ = b.phi(v3)
+                        }
+                    }
+                }
+            }
+        }
+        let eval = b.loadBuiltin("eval")
+        b.callFunction(eval, withArgs: [code])
+
+        let program = b.finalize()
+
+        let lifted_program = fuzzer.lifter.lift(program)
+
+        let expected_program = """
+        const v0 = `
+            const v3 = 13.37 * 1337;
+            const v4 = \\`
+                const v7 = 13.37 + 1337;
+                const v8 = \\`
```

Comment:
Shouldn't this now be \\\\\\\\` so that evaling v4 still leaves this as a template literal?

Reply:
yes, good catch. Turns out escaping is harder than I thought it was :(
I've fixed this bug and updated the testcases to include nested eval statements.